modules/cnf-topology-aware-lifecycle-operator-precache-feature.adoc

```diff
@@ -0,0 +1,161 @@
+
+:_content-type: PROCEDURE
+[id="talo-precache-start_and_update_{context}"]
+= Creating a ClusterGroupUpgrade CR with pre-caching
+
+The pre-cache feature allows the required container images to be present on the spoke cluster before the update starts.
+
+.Prerequisites
+
+* Install the {cgu-operator-first}.
+* Provision one or more managed clusters.
+* Log in as a user with `cluster-admin` privileges.
+
+.Procedure
+
+. Save the contents of the `ClusterGroupUpgrade` CR with the `preCaching` field set to `true` in the `clustergroupupgrades-group-du.yaml` file:
++
+[source,yaml]
+----
+apiVersion: ran.openshift.io/v1alpha1
+kind: ClusterGroupUpgrade
+metadata:
+  name: du-upgrade-4918
+  namespace: ztp-group-du-sno
+spec:
+  preCaching: true <1>
+  clusters:
+  - cnfdb1
+  - cnfdb2
+  enable: false
+  managedPolicies:
+  - du-upgrade-platform-upgrade
+  remediationStrategy:
+    maxConcurrency: 2
+    timeout: 240
+----
+<1> The `preCaching` field is set to `true`, which enables the {cgu-operator} to pull the container images before starting the update.
+
+. When you want to start the update, apply the `ClusterGroupUpgrade` CR by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f clustergroupupgrades-group-du.yaml
+----
+
+.Verification
+
+. Check if the `ClusterGroupUpgrade` CR exists in the hub cluster by running the following command:
++
+[source,terminal]
+----
+$ oc get cgu -A
+----
++
+.Example output
++
+[source,terminal]
+----
+NAMESPACE          NAME              AGE
+ztp-group-du-sno   du-upgrade-4918   10s <1>
+----
+<1> The CR is created.
+
+. Check the status of the pre-caching task by running the following command:
++
+[source,terminal]
+----
+$ oc get cgu -n ztp-group-du-sno du-upgrade-4918 -o jsonpath='{.status}'
+----
++
+.Example output
++
+[source,json]
+----
+{
+  "conditions": [
+    {
+      "lastTransitionTime": "2022-01-27T19:07:24Z",
+      "message": "Precaching is not completed (required)", <1>
+      "reason": "PrecachingRequired",
+      "status": "False",
+      "type": "Ready"
+    },
+    {
+      "lastTransitionTime": "2022-01-27T19:07:24Z",
+      "message": "Precaching is required and not done",
+      "reason": "PrecachingNotDone",
+      "status": "False",
+      "type": "PrecachingDone"
+    },
+    {
+      "lastTransitionTime": "2022-01-27T19:07:34Z",
+      "message": "Pre-caching spec is valid and consistent",
+      "reason": "PrecacheSpecIsWellFormed",
+      "status": "True",
+      "type": "PrecacheSpecValid"
+    }
+  ],
+  "precaching": {
+    "clusters": [
+      "cnfdb1" <2>
+    ],
+    "spec": {
+      "platformImage": "image.example.io"},
+    "status": {
+      "cnfdb1": "Active"}
+    }
+}
+----
+<1> Displays that the update is in progress.
+<2> Displays the list of identified clusters.
+
+. Check the status of the pre-caching job by running the following command on the spoke cluster:
++
+[source,terminal]
+----
+$ oc get jobs,pods -n openshift-talo-pre-cache
```

Comment:
@yliu127 Please check if it is affected by the name change. Has `openshift-talo-pre-cache` been changed?